### PR TITLE
Parse message field in `PlatformException.message`

### LINKED ISSUE #1708: `PlatformException.message` should be treated like `PlatformException.stacktrace`
### Problem Statement

In theory, only [`PlatformException.stacktrace`](https://api.flutter.dev/flutter/services/PlatformException/stacktrace.html) should contain a stacktrace. However, in practice I see a lot of plugins misusing the [`PlatformException.message`](https://api.flutter.dev/flutter/services/PlatformException/message.html) field for stacktraces.

### Solution Brainstorm

Therefore, it would be nice if the `PlatformException.message` field could be treated like the `PlatformException.stacktrace` field.

### Are you willing to submit a PR?

Yes

### PR DESCRIPTION
**Pull request by ueman**
## :scroll: Description

Fixes #1708

## :green_heart: How did you test it?

Tests

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [x] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: flutter/lib/src/event_processor/android_platform_exception_event_processor.dart [hunk 1]
```diff
@@ -99,60 +113,16 @@ class AndroidPlatformExceptionEventProcessor implements EventProcessor {
...
+            // because Android is only run in the dart:io context.
+            current: e.name == Isolate.current.debugName,
+          ),
+        )
         .toList(growable: false);
   }
-
-  /// Remove the StackTrace from [PlatformException] so the message on Sentry
-  /// looks much better.
-  List<SentryException>? _removePlatformExceptionStackTraceFromValue(
```
Comment (ueman) [inline flutter/lib/src/event_processor/android_platform_exception_event_processor.dart:113]: I removed this part of the code as it breaks PlatformException subclasses.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 ### Features
 
+- StackTraces in `PlatformException.message` will get nicely formatted too
```
Comment (ueman) [inline CHANGELOG.md]:
```suggestion
- StackTraces in `PlatformException.message` will get nicely formatted too when present ([#1716](https://github.com/getsentry/sentry-dart/pull/1716))
```

**Review by buenaflor — APPROVED** [review]
nice 👍

**Comment by ueman** [comment]
Feel free to merge it, since I'm unable to do so as an outside contributor :D

**Pull request merged**